specs/PermissionManagement.md

```diff
@@ -0,0 +1,554 @@
+        // the file or folder picker to the end user, and then request "readwrite"
+        // permission for the user's selection.
+        FileEditing = 8,
+
+        // Indicates permission to play audio and video automatically on sites. This
+        // permission affects the autoplay attribute and play method of the audio and
+        // video HTML elements, and the start method of the Web Audio API. See the
+        // [Autoplay guide for media and Web Audio APIs](https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide)
+        // for details.
+        Autoplay = 9,
+
+        // Indicates permission to use fonts on the device. Permission is requested when
+        // developers use the [Local Font Access API](https://wicg.github.io/local-font-access/)
+        // to query the system fonts available for styling web content.
+        LocalFonts = 10,
+    };
+}
+
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2PermissionRequestedEventArgs
+    {
+        // Other members not shown.
+
+        // The permission state set from the `PermissionRequested` event is persisted
+        // across sessions by default and becomes the new default behavior for future
+        // `PermissionRequested` events. Browser heurisitics can affect whether the
+        // event continues to be raised when the state is persisted. Set the
+        // `ShouldPersist` property to `FALSE` to not persist the state beyond the
+        // current request, and to continue to receive `PermissionRequested` events
+        // for this origin and permission kind.
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2PermissionRequestedEventArgs3")]
+        {
+            Boolean ShouldPersist { get; set; };
+        }
+    }
+}
+
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Profile
+    {
+        // Other members of CoreWebView2Profile not shown.
+
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Profile6")]
+        {
+            // Sets permission state for the given permission kind and origin
+            // asynchronously. The change persists across sessions until it is
+            // changed by another call to `SetPermissionState`, or by setting
+            // the `State` property in `PermissionRequestedEventArgs`. The
+            // origin should have a valid scheme and host (e.g.
+            // "https://www.example.com"), otherwise the method fails.
+            // Additional URI parts like path and fragment are ignored. For
+            // example, "https://wwww.example.com/app1/index.html/" is treated
+            // the same as "https://wwww.example.com". See the
+            // [MDN origin definition](https://developer.mozilla.org/en-US/docs/Glossary/Origin) for more details.
+            Windows.Foundation.IAsyncAction SetPermissionStateAsync(
+                CoreWebView2PermissionKind PermissionKind,
+                String origin,
```

Comment:
Since this is structured, should it be something other than a freeform string, or is this consistent with developer expectation?  (string may be correct.  I see the Uri property on CoreWebView2PermissionRequestedEventArgs is string type.)

Reply:
It's required to be a Uri so should be  a Uri type?